cmd/helm/search.go

```diff
@@ -58,6 +61,10 @@ func searchChartRefsForPattern(search string, chartRefs map[string]*repo.ChartRe
 
 func searchCacheForPattern(dir string, search string) ([]string, error) {
 	fileList := []string{}
+	if _, err := os.Stat(dir); err != nil {
+		return fileList, errors.New("No cache exists. Did you run 'helm init'?")
```

Comment:
Another thing we could do is have a method that checks if 'helm init' has been ran and just call that from the commands before proceeding? That way we don't have to handle the special cases in these commands.